ms2rescore/peptideshaker.py

```diff
@@ -48,33 +59,155 @@ def drop_invalid_amino_acids(self, invalid_amino_acids=r"[BJOUXZ]"):
                 invalid_amino_acids
             )
             self._obj = self._obj.drop(index=to_drop)
+    
+    @staticmethod
+    def _get_RawModLocProb(prob_ptm_score):
+        probs = re.findall("\d+\.\d+", prob_ptm_score)
+        if probs:
+            return max([float(x) for x in probs])
+        return 0
+
+    @staticmethod       
+    def _cleanup_protein_ids(prot_ids):
+        # TODO this is too specific to our pipline (OpenProt library)
+        clean_prot_ids = []
+        for prot_id in prot_ids.split(', '):
+            decoy = '_REVERSED' in prot_id
+            prot_acc = prot_id.split('|')[0]
+            if decoy:
+                prot_acc += '_REVERSED'
+            clean_prot_ids.append(prot_acc)
+        return ';'.join(clean_prot_ids)
+
+    def df_from_all_psms(all_psms):
+        all_algos = list(set([x[0] for y in [pd.DataFrame.ext_psm_ann_report._parse_algo_scores(psm['psm_attrs']['Algorithm Score']) for psm in all_psms.values()] for x in y ]))
+        df = []
+        for spec_id, psm in all_psms.items():
+            psm_attrs = psm['psm_attrs']
+            peak_anns = [p for p in psm['peak_anns'] if p['Subtype'] != 'Prec']
+            if not peak_anns:
+                continue # some spectra have no peaks annotated... skip'em

```

Comment:
We should talk about this one. It may be specific to peptideshaker, but it is weird that some spectra with no peaks annotated /or onlt prec peaks annotated are reported...